src/serde/read_cache_lookup.rs

```diff
@@ -28,10 +56,10 @@ pub struct ReadCacheLookup {
     /// the tree hashes of the contents on the left and right
     read_stack: Vec<(Bytes32, Bytes32)>,
 
-    count: HashMap<Bytes32, u32>,
+    count: HashMap<Bytes32, u32, IdentityHash>,
 
     /// a mapping of tree hashes to `(parent, is_right)` tuples
-    parent_lookup: HashMap<Bytes32, Vec<(Bytes32, u8)>>,
+    parent_lookup: HashMap<Bytes32, Vec<(Bytes32, bool)>, IdentityHash>,
```

Comment:
I'd be surprised to see that this change makes any difference.

Reply:
I wonder if you could benchmark with & without this change. If it doesn't improve anything, I think it reads nicer with 0s and 1s than trues and falses.

Reply:
The `IdentityHash` makes a difference. The CPU cost of this function was dominated by the hashing for the `HashSet` and `HashMap`s. (after this patch, it's dominated by growing the `HashMap`.

The `u8` -> `bool` change was because `BitVec` uses `bool`. It emulates the interface of `Vec<bool>`.